allure-testng-adaptor/src/main/java/ru/yandex/qatools/allure/testng/AllureTestListener.java

```diff
@@ -233,4 +243,65 @@ String getSuiteUid(ITestContext iTestContext) {
         }
         return uid;
     }
+
+    /**
+     * Creates test case parameters in XML in case of parametrized test (see TestNG @DataProvider).
+     * Java 8 allows to keep test method parameter names in class files (needs javac -parameters compilation).
+     * In Java 7 default parameter naming is used (arg0, arg1, ... ).
+     *
+     * @param iTestResult parameter passed to onTestStart() callback by TestNG
+     */
+    private void fireAddParameterEvents(ITestResult iTestResult) {
+        Object[] parameters = iTestResult.getParameters();
+        ITestNGMethod method = iTestResult.getMethod();
+        Method nativeMethod = method.getConstructorOrMethod().getMethod();
+
+        boolean atLeastJava8 = isAtLeastJava8();
+        List<String> parameterNames = null;
+        if (atLeastJava8) {
+            parameterNames = getParameterNames(nativeMethod);
+        }
+
+        for (int i = 0; i < parameters.length; i++) {
+            String name = "arg" + i;
+            // overwriting default parameter names for Java >= 8
+            if (atLeastJava8 && !parameterNames.isEmpty()) {
+                name = parameterNames.get(i);
```

Comment:
That is why `!parameterNames.isEmpty()` check. It will be empty if any exception getting method parameter names. `IndexOutOfBoundsException` is also possible if TestNG returns different count of parameters than Java. That would be TestNG library bug, wouldn't it?